modules/qortexRtdProvider.js

```diff
@@ -370,6 +405,31 @@ function shouldAllowBidEnrichment() {
   return true
 }
 
+/**
+ * Passes message out to external page through postMessage method
+ * @param {string} msg message string to be passed to CX-BID-ENRICH target on current page
+ */
+function postBidEnrichmentMessage(msg) {
+  window.postMessage({
+    target: 'CX-BID-ENRICH',
+    message: msg
+  }, window.location.protocol + '//' + window.location.host);
+  logMessage('Message post :: Qortex contextuality has been not been indexed.')
+}
+
+/**
+ * Receives messages passed through postMessage method to QORTEX-PREBIDJS-RTD-MODULE on current page
+ * @param {Object} evt data object holding Event information
+ */
+export function windowPostMessageReceived(evt) {
+  const data = evt.data;
+  if (typeof data.target !== 'undefined' && data.target === 'QORTEX-PREBIDJS-RTD-MODULE') {
+    if (data.message === 'CX-BID-ENRICH-INITIALIZED') {
+      requestContextData()
```

Comment:
should this check `shouldAllowBidEnrichment` prior to requesting? this same check occurs when your module initializes and attempts to retrieve contextual data from your endpoint

Reply:
Great callout! That message would actually never be sent without the check also being done on the other end, but it could still be forced and trigger code that technically shouldn't execute. shouldAllowBidEnrichment() added at message receipt.